app/models/user.rb

```diff
@@ -81,5 +88,20 @@ def search_by_name(query)
 
       left_outer_joins(:profile).where('LOWER(profiles.name) LIKE LOWER(?)', "%#{query}%")
     end
+
+    def from_omniauth(auth)
+      user = where(provider: auth.provider, uid: auth.uid).first_or_initialize do |user|
+        user.email = auth.info.email
+        user.password = Devise.friendly_token[0, 20]
+        user.skip_confirmation!
+      end
+      return user if user.persisted?
+      user.profile.first_name = auth.info.first_name
```

Comment:
`first_name` and `last_name` are not being used throughout the app so you can safely omit it here.